intermediate-js.md

```diff
@@ -3,37 +3,66 @@
 ### Without using Google answer the following:
 
 1. What is the difference between .map() and .filter()?
+.map() loops through items in an array, and .filter() reduces the items in an array
 
 2. Why would you use object destructuring?
+So you don't need to type this. as many times
 
 3. What is the difference between var, let, and const?
+Scope and protection of a variable. If you use const the variable can't be changed later. Let is most often used in local scope while var and const are used in both global and local scope.
 
 4. Why is testing important?
+To identify and eliminate bugs in your code before you launch it.
 
 5. What is a higher order function?
+A function that takes or returns a function.
 
 6. What is the difference between a class and an object?
+A class is a recipe for cooking an object, and an object is an instance of its class.
 
 7. What did you learn during the group project this week? Please include any additional feedback you may have.
 
 
 ### HTML/CSS Review questions: First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. How do you link a CSS file to your HTML page?
+In the head, <link rel="stylesheet" href="filename.css">
 
 2. What is the difference between a div and a span?
+A div is a divider and has many uses. A span is typically used to wrap an html element or text snippet that's going to be styled differently than its fellows, such as a section of text within a p that you want to make a different color.
```

Comment:
You've made a good distinction here! Do some research and familiarize yourself with the concept of inline and block level elements.